python/sglang/srt/function_call/json_detector.py

```diff
@@ -0,0 +1,76 @@
+from typing import List
+
+from sglang.srt.entrypoints.openai.protocol import Tool
+from sglang.srt.function_call.base_format_detector import BaseFormatDetector
+from sglang.srt.function_call.core_types import StreamingParseResult
+
+
+class JsonDetector(BaseFormatDetector):
+    """
+    Detector for JSON array tool calls when JSON schema constraints are active.
+    
+    This detector is used when tool_choice="required" or a specific tool is named,
+    bypassing model-specific parsers in favor of direct JSON array parsing.
+    """
+
+    def __init__(self):
+        super().__init__()
+        # Configure for JSON array parsing
+        self.bot_token = "["
+        self.eot_token = "]"
+        self.tool_call_separator = ", "
+
+    def has_tool_call(self, text: str) -> bool:
+        """
+        Check if the given text contains a JSON tool call (array or single object).
+        
+        Args:
```

Comment:
nit: we can remove Args and Returns in this file as the detailed info is already included in BaseFormatDetector

Args and Returns are useful when the logic is complex, here it is pretty straightforward